src/main/java/org/jasig/portlet/announcements/controller/UserAgentViewNameSelector.java

```diff
@@ -52,10 +59,16 @@ public String select(PortletRequest req, String baseViewName) {
         String userAgent = req.getProperty("user-agent");
         if (userAgent != null && patterns.size() != 0) {
             for (Map.Entry<Pattern,String> y : patterns.entrySet()) {
-                if (y.getKey().matcher(userAgent).matches()) {
-                    rslt.append(y.getValue());
+                if (isRespondr == true) { // normal view format is displayed

```

Comment:
It would be more efficient to test prior to looping since you are basically doing a no-op operation if respondr.